src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -106,24 +142,127 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
                 unsupportedBrowserDetected: () => {
                     deferred.reject(new UnsupportedBrowserError());
                 },
-                cardNonceResponseReceived: (errors, nonce) => {
-                    this._cardNonceResponseReceived(errors, nonce);
+                cardNonceResponseReceived: (errors?: NonceGenerationError[],
+                                            nonce?: string,
+                                            cardData?: CardData,
+                                            billingContact?: Contact,
+                                            shippingContact?: Contact
+                    ) => {
+                    if (errors) {
+                        const error = errors[0];
+
+                        throw new StandardError(error.message);
+                    }
+
+                    if (cardData && cardData.digital_wallet_type !== DigitalWalletType.none) {
+                        this._setExternalCheckoutData(cardData, nonce)
+                            .then(() => this._paymentInstrumentSelected(methodId))
+                            .then(() => squareOptions.onPaymentSelect && squareOptions.onPaymentSelect());
+                    } else {
+                        this._cardNonceResponseReceived(nonce, errors);
```

Comment:
We are passing `errors` in this method to reject the `_deferredRequestNonce` promise.

```js
if (nonceGenerationErrors) {
    this._deferredRequestNonce.reject(nonceGenerationErrors);
}
```

Since the first line in `cardNonceResponseReceived` will throw when there are some errors, I feel like the `_deferredRequestNonce` promise will never get rejected.